src/mscorlib/src/System/Number.cs

```diff
@@ -721,6 +721,78 @@ internal unsafe static Double ParseDouble(String value, NumberStyles options, Nu
             return d;
         }
 
+        internal unsafe static Int32 ParseInt32Fast(String s, NumberFormatInfo info)
+        {
+            if (s == null)
+            {
+                throw new ArgumentNullException(nameof(String));
+            }
+
+            const int StateSign = 1 << 0;
+            const int StateNegative = 1 << 1;
+            const int StateDigits = 1 << 2;
+
+            var state = 0;
+            var ret = 0;
+            fixed (char* sptr = s)
+            {
+                var len = s.Length * 2;
+                var cptr = sptr;
+                while (len > 0)
+                {
+                    len -= 2;
+                    var c = *cptr;
+                    if (c >= '0' && c <= '9' && (state & StateDigits) == 0)
+                    {
+                        state |= StateSign;
+                        ret = ret * 10 + (c - '0');
+                    }
```

Comment:
applied, thanks. The asm code looks better but it doesn't give much in a benchmark.